File: boat-sample/src/test/java/sample/java/xyz/srclab/common/egg/EggSample.java
package sample.java.xyz.srclab.common.egg;

import org.testng.annotations.Test;
import xyz.srclab.common.egg.BoatEggManager;
import xyz.srclab.common.egg.Egg;
import xyz.srclab.common.egg.EggManager;

public class EggSample {

    @Test
    public void testEgg() {
        EggManager eggManager = BoatEggManager.INSTANCE;
        Egg egg = eggManager.pick("Hello, Boat Egg!");
        egg.hatchOut("出来吧，神龙！");
    }
}

File: boat-sample/src/test/kotlin/sample/kotlin/xyz/srclab/common/egg/EggSample.kt
package sample.kotlin.xyz.srclab.common.egg

import org.testng.annotations.Test
import xyz.srclab.common.egg.BoatEggManager

class EggSample {

    @Test
    fun testEgg() {
        val egg = BoatEggManager.pick("Hello, Boat Egg!")
        egg.hatchOut("出来吧，神龙！")
    }
}